### ISSUE TITLE
How to secure superset charts embedded using iframes into public websites

### ISSUE THREAD

**Issue by akmr9** [apache/superset#17759]
Hello,

I am new to superset. Can someone please guide me on how to secure the superset charts embedded into public websites using iframes? Basically I want to restrict the usage of the charts outside of my website. Can we use host application authentication to authenticate with superset or can we use csrf tokens etc?

**Comment by przemekd** [comment]
Yep, you can implement a custom authentication module (see https://sairamkrish.medium.com/apache-superset-custom-authentication-and-integrate-with-other-micro-services-8217956273c1).

**Comment by aeperalta** [comment]
There is [SIP-75](https://github.com/apache/superset/issues/17187) that is gonna be released the next year Q1-Q2 I readed in [slack](https://apache-superset.slack.com/archives/C01EP56QGTS/p1637830159042800). And there is the solution that @przemekd said. Personally, I'll wait for the "official" authentication embedding.
